### PR TITLE
added some documentation for using regtest

### PR DESCRIPTION
**Pull request by philbertw4**
my git foo skills are not great yet, but my goal here is to add some documentation for how people might go about actually using regtest locally with their defiads setup. Github seems to indicate that the best way for me to do this is to open a pull request here on @notmandatory 's fork and the changes should show up in our #22 pull request pertaining to defiads issue #18

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -60,7 +60,18 @@ negotiation and thereby give rise to bitcoin's first truly risk less interest ra
...
-# Run
+
+# Running On Local Regtest Network (Recommended for developers)
+1. Be sure to have a somewhat recent version of bitcoin core installed.
+2. Start bitcoin core in regtest mode `$ bitcoind -regtest -daemon`.
+3. Your local bitcoin instance will now be listening for connections on (default) port 18444
+4. Whenever blocks need to be generated for your testing, you can use the command `$ bitcoin-cli -testnet generatetoaddress <address for mined coins to be sent>`. If you do not know how to use this command, `$ bitcoin-cli help <command>` is your friend. As a reminder, coins mined in block `n` are spendable in block `n+100`.
+5. Now you can run your local defiads node and have it connect to your local regtest network:
+```
+$ target/release/defiads --bitcoin-network regtest --bitcoin-peers 127.0.0.1:18444
```
Comment (notmandatory) [inline README.md]: I think it's worth adding the  "--bitcoin-discovery OFF" option

**Comment by notmandatory** [comment]
Also this MR is to my folks master branch, I think you need to change it to be an MR to my fix_regtest_check branch.. if github will let you do that

**Comment by notmandatory** [comment]
oh and one other important thing, as far as I know defiads will only work with the bip157-net branch of the bitcoin code since only this version serves bip157 block filters.  How do you feel about just pointing to my [docker-compose regtest project](https://github.com/notmandatory/bitcoin-regtest-box)? or we can add those three files to the defiads project under a contrib/regtest directory.

**Comment by notmandatory** [comment]
@philbertw4 on second thought, I think we should do this as a new PR, separate from the code network assert changes, especially if we add the regtest docker-compose stuff.  It's usually best to not try to do too many things in one PR so that the review is easier.

**Comment by philbertw4** [comment]
Good points all the way around. We can add the docs as a separate PR.
Regarding whether to use your docker regtest project or not, I am pretty
indifferent because my docker experience is minimal. If we can do what you
suggest and add the necessary files to defiads repository so things are
nice and self contained, that makes sense to me. However, someone with more
experience/preferences than me should probably weigh in on it.

**Comment by philbertw4** [comment]
Ok, now to see if I can figure out how to change these documentation-related commits over to a different separate pull request. Standby.

**Pull request closed without merging**